frontend/src/stores/GameStore.ts

```diff
@@ -31,30 +36,74 @@ class GameStore {
     makeAutoObservable(this);
   }
 
-  placeTile(row: number, column: number) {
+  placeTile(row: number, column: number, fromWebsocket: boolean) {
     const tileToChange = this.boardState.find((tile) => tile.row === row && tile.column === column);
     console.log(`tile to change: `, tileToChange);
     if (tileToChange && this.tileInHand) {
       if (validateTilePlacement(row, column)) {
         tileToChange.state = TileState.TAKEN;
         tileToChange.tile = this.tileInHand;
+        if (!fromWebsocket) {
+          const websocketMessageParser = new WebsocketMessageParser();
+          const tilePlacementMessage = new TilePlacementMessage('');
+          tilePlacementMessage.id = this.tileInHand.id;
+          tilePlacementMessage.row = row;
+          tilePlacementMessage.column = column;
+          tilePlacementMessage.rotation = this.tileInHand.rotation;
+
+          websocketMessageParser.parse(tilePlacementMessage, WebSocketEvent.SEND_TILE_PLACED);
+
+          socket.emit(
+            WebSocketEvent.SEND_TILE_PLACED,
+            `${this.tileInHand.id}_${row}_${column}_${this.tileInHand.rotation}`,
+          );
+        }
+
```

Comment:
placeTIle method is getting quite long. Please extract the body of this if statement to the separate function. eg. emitTilePlacementMessage

Reply:
done